### ISSUE TITLE
Ask for "Reload Obsidian" when it is safe to do

### ISSUE THREAD

**Issue by alx-xlx** [vrtmrz/obsidian-livesync#413]
Whenever I Customization Sync the device the "Reload Obsidian" options shows randomly when important sync is happening, And and when pressing the reload button it just corrupts the sync and it fails.

I think it would be a better thing if it asks for Reload when all important sync is over and its safe to reload.

**Comment by vrtmrz** [comment]
I appreciate your proposals! Indeed reloading while busy states may be potentially harmful.
And, a while ago, we could not detect the busyness, but now it can be with a bit more certainty.

This might be a good opportunity. I will try to implement this!

**Comment by vrtmrz** [comment]
It has been implemented at v0.23.7. Would you mind if I ask you to check the behaviour, please?

**Comment by alx-xlx** [comment]
Yes, it works well. Now we don't have to wait and constantly stare at the indicators to restart obsidian. :p

**Comment by alx-xlx** [comment]
Just curious if the busyness status of livesync can be accessed by other plugins ?

**Comment by vrtmrz** [comment]
Sorry for being late!
Actually, I also wanted the feature for interop between the plugins but nothing there now.
For now, we are only able to access Self-hosted LiveSync by accessing `app.plugins.plugins["obsidian-livesync"]` and its props.

If you have specific scenarios in mind, I am happy to explore ways to make communication a bit safer. For instance, I could register custom event handlers in the window to respond to specific requests.
